docs/src/building.md

```diff
@@ -0,0 +1,66 @@
+<!--
+SPDX-FileCopyrightText: 2021 Daniel Fullmer and robotnix contributors
+SPDX-License-Identifier: MIT
+-->
+
+# Building
+
+## Build outputs
+
+Robotnix provides a number of Nix outputs that can be built with a given configuration.
+The specific output is selected using the `-A` option of `nix-build`.
+For example, the following will build an `img` zip associated with the configuration in `config.nix`.
+```shell
+$ nix-build --arg configuration ./config.nix -A img
+```
+
+Some of the outputs provided by robotnix are the following:
+- `img` - Image zip, which can be flashed to a device using `fastboot update`.
+- `factoryImg` - Factory image, a zip which contains the contents of `img` as well as a radio / bootloader if available.
+- `ota` - Over-the-air zip, which can be flashed to a device in recovery mode using `adb sideload`.
+- `releaseScript` - Script which produces the `img`, `ota`, and `factoryImg` products outside of Nix.
+- `generateKeysScript` - Script to generate required device / application keys for a given configuration.
+
+## Building and Signing Releases
+After creating a configuration file, you need to generate keys for your device (if you are using signed builds, with `signing.enable = true;`):
+```console
+$ nix-build --arg configuration ./crosshatch.nix -A generateKeysScript -o generate-keys
+$ ./generate-keys ./keys
+```
+This will create a `keys` directory containing the app and device keys needed for the build.
+The output of this script should be placed in the location specified by `keyStorePath` in the robotnix configuration.
+
+Sometimes changing your configuration will require that you generate additional new keys (e.g. for additional applications).
+Rebuilding and rerunning the generate keys script will produce the new keys (without overwriting your existing keys).
+
+Next, build and sign your release.
+There are two ways to do this.
+The first option is to build the final products entirely inside Nix.
+```console
+$ nix-build ./default.nix --arg configuration ./crosshatch.nix -A img --option extra-sandbox-paths /keys=$(pwd)/keys
+```
+This, however, will require a nix sandbox exception so the secret keys are available to the build scripts.
+To use `extra-sandbox-paths`, the user must be a `trusted-user` in `nix.conf`.
+Additionally, the nix builder will also need read access to these keys.
+This can be set using `chgrp -R nixbld ./keys` and `chmod -R g+r ./keys`.
+
+The second option involves creating a "release script" which does the final build steps of signing target files and creating `img`/`ota` files outside of Nix:
+```console
+$ nix-build --arg configuration ./crosshatch.nix -A releaseScript -o release
+$ ./release ./keys
+```
```

Comment:
Maybe mention that in this case fingerprint might have to be set manually, because some things (e.g. Auditor) need to know them during the build.

Reply:
I will actually test this soon, because I want to move to distributed robotnix builds myself.

Reply:
Added a note. 45f484f85175883a332d9a8935fcac6ed88097a0

Technically, I think you can still use `keyStorePath` to automatically get fingerprints even when using `releaseScript`.  It would require that you have the public keys available through the `keyStorePath` on the remote machine.